src/common/CudaDataUtils.hpp

```diff
@@ -42,6 +42,31 @@ __device__ __forceinline__ unsigned long long device_timer()
   return global_timer;
 }
 
+/*!
+ * \brief Method for launching a CUDA kernel with given configuration.
+ *
+ *        Note: method includes a call to check whether number of args
+ *              in kernel signature matches number of args passed to this 
+ *              method.
+ */
+template <typename... Args, typename F = void (*)(Args...)>
+void RPlaunchCudaKernel(F kernel,
+                        const size_t& numBlocks, const size_t& dimBlocks,
```

Comment:
```suggestion
                        const dim3& numBlocks, const dim3& dimBlocks,
```

Reply:
is that an implicit type conversion?

Reply:
We do launch with dim3 instead of a single integer sometimes. There is an implicit conversion to dim3 from integers, yes.